src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+/**
+ * The Parser class that makes sense of the user input.
+ */
+public class Parser {
+    private String input;
+    private String[] inputArr;
+    private String command;
+
+    /**
+     * Constructor for a Parser object.
+     *
+     * @param input The user input to be parsed.
+     */
+    public Parser(String input) {
+        this.input = input;
+        this.inputArr = input.split(" ");
+        this.command = inputArr[0];
+    }
+
+    private boolean isEditingTask() {
+        return command.equals("done") || command.equals("delete");
+    }
+
+    private boolean isFindingTask() {
+        return command.equals("find");
+    }
+
+    private boolean isAddingNewTask() {
+        return command.equals("todo") || command.equals("deadline") || command.equals("event");
+    }
+
+    private String[] getAddTaskArgs() throws DukeException {
```

Comment:
Maybe you could use switch statements instead of a lot of if else in order to make the code more readable